### PR TITLE
Index fill pytorch

### PR DESCRIPTION
**Pull request by muhd360**
### Details:
added Support  for aten::index_fill_ for pytorch models
 - *...*

### Tickets:
23326

### CODE REVIEW

File: src/frontends/pytorch/src/op/index_fill_.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include "utils.hpp"
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_index_copy_(const NodeContext& context) {
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/index_fill_.cpp:26]:
```suggestion
OutputVector translate_index_fill_(const NodeContext& context) {
```

**Review by mvafin — CHANGES_REQUESTED** [review]
You need to add the conversion function in the list of operators here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/pytorch/src/op_table.cpp

**Comment by muhd360** [comment]
> You need to add the conversion function in the list of operators here: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/pytorch/src/op_table.cpp

ok will do

**Review by muhd360 — COMMENTED** [review]
pls check

File: src/frontends/pytorch/src/op_table.cpp [hunk 2]
```diff
@@ -483,6 +483,7 @@ const std::unordered_map<std::string, CreatorFunction> get_supported_ops_ts() {
         // aten::imag - Supported in limited set of patterns
         // aten::index - Supported in limited set of patterns
         {"aten::index_copy_", op::inplace_op<op::translate_index_copy_>},
+        {"aten::index_fill_", op::inplace_op<op::translate_index_fill_>},
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:486]: Also need to declare it with `OP_CONVERTER` macros

Comment (muhd360) [reply src/frontends/pytorch/src/op_table.cpp:486]: ok sure

Comment (muhd360) [reply src/frontends/pytorch/src/op_table.cpp:486]: sorry as its my first pr im ill informed

Comment (muhd360) [reply src/frontends/pytorch/src/op_table.cpp:487]: hi pls review

**Comment by muhd360** [comment]
hi,how can i address these issues??

**Comment by mvafin** [comment]
> hi,how can i address these issues??

Yes, please fix code style issues: https://github.com/openvinotoolkit/openvino/actions/runs/10785003751/job/29925825017
And build issues

**Comment by muhd360** [comment]
pls verify @mvafin

**Comment by muhd360** [comment]
hi,i fixed the issues but still the same error is popping up???

**Comment by muhd360** [comment]
@mvafin anything i can do to fix this error??

**Comment by muhd360** [comment]
> @mvafin anything i can do to fix this error??

??

**Comment by muhd360** [comment]
@mlukasze what seems to be the error i cant pinpoint it pls help

**Comment by mlukasze** [comment]
I'm looking on it, will back to you if it is something you could fix.

**Comment by muhd360** [comment]
thanks for your response

**Comment by muhd360** [comment]
hi @mlukasze when will this be merged??

File: src/frontends/pytorch/src/op/index_fill_.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+namespace op {
+
+OutputVector translate_index_fill_(const NodeContext& context) {
+    num_inputs_check(context, 4, 4);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+    auto index = context.get_input(2);
+    auto value = context.get_input(3);
+
+    auto const_1_vec = v0::Constant::create(element::i32, Shape{1}, {value});
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/index_fill_.cpp:33]: `value` is Node, you are trying to create a constant from it.

**Review by mvafin — CHANGES_REQUESTED** [review]
Your code can't be compiled

File: src/frontends/pytorch/src/op/index_fill_.cpp (hunk 3 above)
Comment (muhd360) [reply src/frontends/pytorch/src/op/index_fill_.cpp:33]: ok will fix

**Comment by muhd360** [comment]
@mvafin hi pls  check now

**Comment by mvafin** [comment]
@muhd360 Tests fail with:
```
E       Exception happened during conversion of operation aten::index_fill_ with schema aten::index_fill_.int_Tensor(Tensor(a!) self, int dim, Tensor index, Tensor value) -> Tensor(a!)
E       Check 'backward_compatible_check || in_out_elements_equal' failed at src/core/shape_inference/include/reshape_shape_inference.hpp:352:
E       While validating node 'opset1::Reshape Reshape_7527163 (opset1::Mod aten::index_fill_/Mod[0]:i64[], opset1::Constant Constant_7527154[0]:i32[1]) -> (dynamic[...])' with friendly_name 'Reshape_7527163':
E       Shape inference input shapes {[],[1]}
E       Requested output shape [0] is incompatible with input shape
```

**Comment by muhd360** [comment]
ok working on it

**Comment by muhd360** [comment]
@mvafin im getting this error now and i cant seem to figure out the solution
Screenshot from 2024-09-27 09-40-06

**Comment by mvafin** [comment]
@muhd360 This error is printed if you do not have implementation of the conversion for this operation. Do you have a release version of openvino in your enviroment?

**Comment by mvafin** [comment]
> but i did add the converter right?yes i have openvino installed on the system

You need to build openvino with your changes, do not use release version.

**Comment by muhd360** [comment]
how can i do this?

**Comment by mvafin** [comment]
@muhd360 Please refer here https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md

**Comment by muhd360** [comment]
thanks a ton

**Pull request closed without merging**